lgc/builder/YCbCrConverter.cpp

```diff
@@ -332,10 +345,21 @@ void YCbCrConverter::genSamplerDescChroma() {
 // Generate image descriptor for chroma channel
 void YCbCrConverter::genImgDescChroma() {
   SqImgRsrcRegHandler proxySqRsrcRegHelper(m_builder, m_imgDescLuma, m_gfxIp);
-  Value *width = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Width);
-  Value *height = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Height);
-  m_width = m_builder->CreateUIToFP(width, m_builder->getFloatTy());
-  m_height = m_builder->CreateUIToFP(height, m_builder->getFloatTy());
+
+  Value *width = nullptr;
+  Value *height = nullptr;
+
+  if ((m_metaData.word5.lumaDepth > 1) && (m_metaData.word1.planes > 1)) {
+    m_width = ConstantFP::get(m_builder->getFloatTy(), m_metaData.word4.lumaWidth);
+    m_height = ConstantFP::get(m_builder->getFloatTy(), m_metaData.word4.lumaHeight);
+    width = m_builder->CreateFPToUI(m_width, m_builder->getInt32Ty());
```

Comment:
I prefer to get int constant from m_metaData.word4.lumaWidth/lumaHeight directly, instead of add convert instruction